src/main/java/com/twentyonec/ItemsLogger/utils/Storage.java

```diff
@@ -0,0 +1,261 @@
+
+		config.addDataSourceProperty("cachePrepStmts", "true");
+		config.addDataSourceProperty("prepStmtCacheSize", "250");
+		config.addDataSourceProperty("prepStmtCacheSqlLimit", "2048");
+		config.addDataSourceProperty("useServerPrepStmts", "true");
+		config.addDataSourceProperty("rewriteBatchedStatements", "true");
+		config.addDataSourceProperty("maintainTimeStats", "false");
+
+		config.setMaximumPoolSize(10);
+		config.setMinimumIdle(10);
+		config.setIdleTimeout(300000);
+		config.setMaxLifetime(600000);
+		config.setConnectionTimeout(5000);
+		config.setInitializationFailTimeout(-1);
+
+		try {
+			this.dataSource = new HikariDataSource(config);
+		} catch (Exception e) {
+			e.printStackTrace();
+		}
+
+	}
+
+	public Connection getConnection() throws SQLException {
+		return this.dataSource.getConnection();
+	}
+
+	public void update(final String sql) {
+
+		try (Connection connection = this.getConnection()) {
+
+			PreparedStatement statement = connection.prepareStatement(sql);
+
+			plugin.debugMessage("Preparing statement for update.");
+			statement.executeUpdate();
+			plugin.debugMessage("Successfully executed update statement. (" + sql + ")");
+
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute update statement. (" + sql + ")");
+			e.printStackTrace();
+		}
+
+	}
+
+	public void update(String sql, Date date, Time time) {
+
+		try (Connection connection = this.getConnection()) {
+
+			PreparedStatement statement = connection.prepareStatement(sql);
+			statement.setDate(1, date);
+			statement.setTime(2, time);
+
+			plugin.debugMessage("Preparing statement for update.");
+			statement.executeUpdate();
+			plugin.debugMessage("Successfully executed update statement. (" + sql + ")");
+
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute update statement. (" + sql + ")");
+			e.printStackTrace();
+		}
+
+	}
+
+	public ResultSet query(final String sql) {
+
+		try (Connection connection = this.getConnection()){
+
+			PreparedStatement statement = connection.prepareStatement(sql);
+
+			plugin.debugMessage("Preparing statement for query.");
+			ResultSet resultSet = statement.executeQuery(sql);
+			plugin.debugMessage("Successfully executed query statement. (" + sql + ")");
+
+			return resultSet;
+		} catch (SQLException e) {
+			plugin.debugMessage("Error while attempting to execute query statement. (" + sql + ")");
+			e.printStackTrace();
+			return null;
+		}
+	}
+
+	public void setUpTable() {
+		this.connect();
+
+		plugin.debugMessage("Attempting to set up tables if they do not exist.");
+		final String update = "CREATE TABLE IF NOT EXISTS itemslogger("
+				+ "uuid VARCHAR(36) NOT NULL, "
+				+ "inventory VARCHAR(12000), "
+				+ "cause VARCHAR(255) NOT NULL, "
+				+ "loc_x REAL NOT NULL, "
+				+ "loc_y REAL NOT NULL, "
+				+ "loc_z REAL NOT NULL, "
+				+ "experience REAL NOT NULL, "
+				+ "date DATE NOT NULL, "
+				+ "time TIME NOT NULL);";
```

Comment:
This can go into a file of constants since it never changes. Also that variable name, at least change it to something that describes what it is like `createTableQuery`